kyuubi-main/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -158,4 +134,30 @@ object ProcBuilder {
     }
   }
 
+  private def getRollAppendProcessLogFile(
+      workingDir: Path, module: String, processLogRetainTimeMillis: Long): File = synchronized {
```

Comment:
In the worst case, does this block to visit all files during `processLogRetainTimeMillis`(maybe a day or even longer?)

Reply:
Logically it is. I make a simple stress test offline. Seems it's OK.
```
(1 to 100).foreach { _ =>
          pool.execute(new Runnable {
            override def run(): Unit = {
              val pb = new FakeSparkProcessBuilder(config) {
                override val workingDir: Path = fakeWorkDir
              }
              try {
                val p = pb.start
                p.waitFor()
              } finally {
                pb.close()
              }
            }
          })
        }
```
```
val start = System.currentTimeMillis()
atomicTest()
println(System.currentTimeMillis() - start)
```

That shows only 500ms.

Reply:
Can we just move forward when an error occurs? and track the error count to be under the threshold. Listing plenty of files can increase random errors? through?

Reply:
We won't get a exception since it's a overwrite file. Update the new implementation.